internal/resolution/resolver_test.go

```diff
@@ -19,112 +18,113 @@ import (
 				Name: "prometheus",
 			},
 			Spec: v1alpha1.OperatorSpec{
 				PackageName: "prometheus",
 			},
-		}
-		client := FakeClient(resource)
-		entitySource := FailEntitySource{}
-		resolver := resolution.NewOperatorResolver(client, entitySource)
-		solution, err := resolver.Resolve(context.Background())
-		Expect(solution).To(BeNil())
-		Expect(err).To(HaveOccurred())
-	})
-
-	It("should return an error if the client throws an error", func() {
-		client := NewFailClientWithError(fmt.Errorf("something bad happened"))
-		entitySource := input.NewCacheQuerier(testEntityCache)
-		resolver := resolution.NewOperatorResolver(client, entitySource)
-		solution, err := resolver.Resolve(context.Background())
-		Expect(solution).To(BeNil())
-		Expect(err).To(HaveOccurred())
-	})
-})
+		},
+		&v1alpha1.Operator{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "packageA",
+			},
+			Spec: v1alpha1.OperatorSpec{
+				PackageName: "packageA",
+			},
+		},
+	}
+
+	for _, tt := range []struct {
+		Name              string
+		Client            client.Client
+		EntitySource      input.EntitySource
+		SelectedVariables []deppy.Identifier
+		ExpectedError     error
+	}{
+		{
+			Name:         "should resolve the packages described by the available Operator resources",
+			Client:       FakeClient(testResource...),
+			EntitySource: testEntitySource,
+			SelectedVariables: []deppy.Identifier{"operatorhub/packageA/2.0.0",
+				"operatorhub/prometheus/0.47.0",
+				"required package packageA",
+				"required package prometheus"},
+			ExpectedError: nil,
+		},
+		{
+			Name:              "should not return an error if there are no Operator resources",
+			Client:            FakeClient(),
+			EntitySource:      testEntitySource,
+			SelectedVariables: []deppy.Identifier{},
+			ExpectedError:     nil,
+		},
+		{
+			Name:          "should return an error if the entity source throws an error",
+			Client:        FakeClient(testResource...),
+			EntitySource:  FailEntitySource{},
+			ExpectedError: fmt.Errorf("error calling filter in entity source"),
+		},
+		{
+			Name:          "should return an error if the client throws an error",
+			Client:        NewFailClientWithError(fmt.Errorf("something bad happened")),
+			EntitySource:  testEntitySource,
+			ExpectedError: fmt.Errorf("something bad happened"),
+		},
+	} {
+		t.Run(tt.Name, func(t *testing.T) {
+			client := tt.Client
+			entitySource := tt.EntitySource
+			resolver := resolution.NewOperatorResolver(client, entitySource)
+			solution, err := resolver.Resolve(context.Background())
+
+			if err != nil {
+				assert.Equal(t, err, tt.ExpectedError)
+				assert.Nil(t, solution)
+			} else {
+				assert.Len(t, solution.SelectedVariables(), len(tt.SelectedVariables))
+				for _, identifier := range tt.SelectedVariables {
+					assert.True(t, solution.IsSelected(identifier))
+				}
+				assert.NoError(t, err)
+			}
+
+		})
+	}
+}
+
+func FakeClient(objects ...client.Object) client.Client {
```

Comment:
It doesn't seem like FakeClient is used anywhere else, this doesn't necessarily need to be exported

Reply:
The FakeClient is being used inside the structure by the Client field for creating fake clients.

Reply:
As long as it's being used within the same package, it doesn't need to be exported